commit 7ebfc8f69a37bb28380c97819fd6437eec37dd8c
Author: Toshi Kani <toshi.kani@hpe.com>
Date:   Fri Sep 14 08:51:14 2018 -0600

    ext2, dax: set ext2_dax_aops for dax files
    
    commit 9e796c9db93b4840d1b00e550eea26db7cb741e2 upstream.
    
    Sync syscall to DAX file needs to flush processor cache, but it
    currently does not flush to existing DAX files.  This is because
    'ext2_da_aops' is set to address_space_operations of existing DAX
    files, instead of 'ext2_dax_aops', since S_DAX flag is set after
    ext2_set_aops() in the open path.
    
    Similar to ext4, change ext2_iget() to initialize i_flags before
    ext2_set_aops().
    
    Fixes: fb094c90748f ("ext2, dax: introduce ext2_dax_aops")
    Signed-off-by: Toshi Kani <toshi.kani@hpe.com>
    Suggested-by: Jan Kara <jack@suse.cz>
    Cc: Jan Kara <jack@suse.cz>
    Cc: Dan Williams <dan.j.williams@intel.com>
    Cc: "Theodore Ts'o" <tytso@mit.edu>
    Cc: Andreas Dilger <adilger.kernel@dilger.ca>
    Cc: <stable@vger.kernel.org>
    Signed-off-by: Jan Kara <jack@suse.cz>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/fs/ext2/inode.c b/fs/ext2/inode.c
index 71635909df3b..b4e0501bcba1 100644
--- a/fs/ext2/inode.c
+++ b/fs/ext2/inode.c
@@ -1448,6 +1448,7 @@ struct inode *ext2_iget (struct super_block *sb, unsigned long ino)
 	}
 	inode->i_blocks = le32_to_cpu(raw_inode->i_blocks);
 	ei->i_flags = le32_to_cpu(raw_inode->i_flags);
+	ext2_set_inode_flags(inode);
 	ei->i_faddr = le32_to_cpu(raw_inode->i_faddr);
 	ei->i_frag_no = raw_inode->i_frag;
 	ei->i_frag_size = raw_inode->i_fsize;
@@ -1517,7 +1518,6 @@ struct inode *ext2_iget (struct super_block *sb, unsigned long ino)
 			   new_decode_dev(le32_to_cpu(raw_inode->i_block[1])));
 	}
 	brelse (bh);
-	ext2_set_inode_flags(inode);
 	unlock_new_inode(inode);
 	return inode;
 	
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1246_7ebfc8f69a37bb28380c97819fd6437eec37dd8c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1246_7ebfc8f69a37bb28380c97819fd6437eec37dd8c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
